File: problem11.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int max_container = 0;
        int temp = 0;
        while( left < right){
            if(height[left] <= height[right]){
                temp = height[left] * (right - left);
                left++;
            }
            else{
                temp = height[right] * (right - left);
                right--;
            }

            if(max_container < temp){
                max_container = temp;
            }
        }
        return max_container;
    }
};

File: problem11.py
#################
## Container with most water
##################

class Solution:
    def maxArea(self, height: List[int]):
        # Two point of list 
        start = 0
        end = len(height)-1
        # The initial value of space is 0
        maxWater = 0
        # Loop until two end points meet
        while start < end:
            if height[start] <= height[end]:
                temp = height[start] * (end - start)
                start += 1
            else:
                temp = height[end] * (end -start)
                end -= 1
            if maxWater < temp:
                maxWater = temp
        return maxWater